File: Axiom/Internal/Ast/WhileStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Axiom.Internal.Ast
{
    internal class WhileStatement : Statement
    {
        public WhileStatement(Expression condition, Statement body)
        {
            Condition = condition;
            Body = body;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public Expression Condition { get; private set; }
        public Statement Body { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/WhileStatement.java
package com.moybl.axiom.ast;

import com.moybl.axiom.Position;

public class WhileStatement extends Statement {

	private Expression condition;
	private Statement body;

	public WhileStatement(Position position, Expression condition, Statement body) {
		super(position);
		this.condition = condition;
		this.body = body;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public Expression getCondition() {
		return condition;
	}

	public Statement getBody() {
		return body;
	}

}
